core/src/test/java/org/apache/calcite/plan/volcano/VolcanoPlannerTest.java

```diff
@@ -296,6 +300,32 @@
                 "PhysSingleRel:RelSubset#0.PHYS.[]")));
   }
 
+  @Test void testTypeMismatch() {
+    VolcanoPlanner planner = new VolcanoPlanner();
+    planner.addRelTraitDef(ConventionTraitDef.INSTANCE);
+    planner.addRule(MockPhysLeafRule.INSTANCE);
+
+    RelOptCluster cluster = newCluster(planner);
+    NoneLeafRel leafRel =
+        new NoneLeafRel(
+            cluster,
+            "a");
+    RelNode convertedRel =
+        planner.changeTraits(
+            leafRel,
+            cluster.traitSetOf(PHYS_CALLING_CONVENTION));
+    planner.setRoot(convertedRel);
+    try {
+      planner.chooseDelegate().findBestExp();
+    } catch (RuntimeException e) {
```

Comment:
Please use `assertThrows` to make sure that an exception is actually thrown.

Reply:
Done. Thank you for your advice.